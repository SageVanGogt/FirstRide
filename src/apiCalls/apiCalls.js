import { geoKey } from './../apiKey';

const signinUser = async (user) => {
  const url = 'http://localhost:3000/api/users/';
  const init = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, init);
  const profile = await response.json();
  return profile;
}; 

const signupUser = async (user) => {
  const url = 'http://localhost:3000/api/users/new';
  const body = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const userRecieved = await response.json();
  return userRecieved;
};

const fetchDestination = async (name) => {
  const url = `http://localhost:3000/api/locations/${name}`;
  const response = await fetch(url);
  const destination = await response.json();
  return destination;
};

const fetchRides = async (destination) => {
  const url = `http://localhost:3000/api/rides/get/${destination}`;
  const body = {
    method: 'POST', 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = await response.json();
  return rides;
};

const fetchPickups = async (location) => {
  const url = `http://localhost:3000/api/pickup/get/${location}`;
  const body = {
    method: "POST", 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const pickups = await response.json();
  return pickups;
};

const fetchGeocode = async (address) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${address}&key=${geoKey}`;  
  try {
    const response = await fetch(url);
    const locationData = await response.json();
    if (!locationData) {
      return new Error('error')
    }
    return locationData;
  } catch (error) {
    throw error;
  }
}; 

const submitNewRide = async (ride) => {
  const url = `http://localhost:3000/api/rides/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(ride),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const rideId = await response.json();
  return rideId;
};

const submitNewPickup = async (location) => {
  const url = `http://localhost:3000/api/pickup/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(location),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const pickupId = await response.json();
  return pickupId;
};

const fetchRidesPassengers = async (locationId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/passengers/${locationId}`;
  const body = {
    method: 'POST', 
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const passengers = await response.json();
  return passengers;
};

const postRidesPassengers = async (newPassenger) => {
  const url = `http://localhost:3000/api/rides_passengers/new`;   
  const body = {
    method: "POST",
    body: JSON.stringify(newPassenger),
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const ridePassenger = await response.json();
  return ridePassenger;
};

const postNewProfile = async (profile) => {
  const url = `http://localhost:3000/api/profiles/new`;
  const body = {
    method: "POST",
    body: JSON.stringify(profile),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const profileInfo = await response.json();
  return profileInfo;
};

const removePassengerRide = async (rideId, passengerId, destinationId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/passengers/${passengerId}/destination/${destinationId}`; 
  const body = {
    method: "DELETE",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const deletedId = await response.json();
  return deletedId;
};

const fetchUserRides = async (userId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/rides/${userId}`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = response.json();
  return rides;
};

const fetchRidesFromUser = async (rideId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/get/user/`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const ridesInfo = await response.json();
  return ridesInfo;
};

const reverseGeoCode = async (lat, lng) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lng}&key=${geoKey}`;
  const response = await fetch(url);
  const geoData = await response.json();
  return geoData;
}

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewRide,
  submitNewPickup,
  fetchRidesPassengers,
  postRidesPassengers,
  postNewProfile,
  removePassengerRide,
  fetchUserRides,
  fetchRidesFromUser,
  reverseGeoCode
};