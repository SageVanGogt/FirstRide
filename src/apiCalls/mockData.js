export const mockUser = {
  id:1,
  user_id:"456",
  user_img:"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
  bio:"I like to eat",
  rating:"Solid 7"
}