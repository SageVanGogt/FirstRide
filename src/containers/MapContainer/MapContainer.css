.availableRidesMap {
  position: absolute;
  height: 100vh;
  width: calc(100vw - 200px);
}

.map-container {
  position: fixed;
  height: 100vh;
  left: 200px;
  width: calc(100vw - 200px);
}