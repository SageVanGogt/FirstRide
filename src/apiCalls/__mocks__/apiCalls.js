import * as MOCK from './../mockData';

const signinUser = jest.fn().mockImplementation(() => 
  Promise.resolve({
    user: [{
    "id":1,
    "user_id":"456",
    "user_img":"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
    "bio":"I like to eat",
    "rating":"Solid 7"
    }]
  })
);

const signupUser = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1
}));

const fetchDestination = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1,
  "location_name": "Red Rocks",
  "location_lat_lng": "12341, 123412"
}));

const fetchRides = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockRides));

const fetchPickups = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockPickups));

const fetchGeocode = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockGeoInfo));

const submitNewRide = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const submitNewPickup = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const postRidesPassengers = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewPickup,
  submitNewRide,
  postRidesPassengers
}