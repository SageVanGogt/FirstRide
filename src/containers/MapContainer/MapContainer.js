
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { mapKey } from './../../apiKey';
import MapComponent from '../../components/MapComponent/MapComponent';
import { connect } from 'react-redux';
import * as actions from './../../actions/pickups';
import { fetchPickups } from '../../apiCalls/apiCalls';
import './MapContainer.css';

const mapUrl = `https://maps.googleapis.com/maps/api/js?key=${mapKey}&v=3.exp&libraries=geometry,drawing,places`;

export class MapContainer extends Component {
  constructor(props) {
    super(props);
  }

  toggleShowing = (pickup) => {
    this.props.toggleShowing(pickup);
  };

  containerElement = () => (<div className="availableRidesMap"/>);
  
  loadingElement = () => (<div style={{ height: `100%` }}/>);
  mapElement = () => (<div style={{ height: `100%` }} />);

  render() {
    return (
      <div className="map-container">
        <MapComponent
          position={this.props.currentLocation}
          googleMapURL={mapUrl}
          rides={this.props.rides}
          markerCoords={this.props.pickupLocations}
          toggleShowing={this.toggleShowing}
          loadingElement={this.loadingElement()}
          containerElement={this.containerElement()}
          mapElement={this.mapElement()} />
      </div>
    );
  }
};

export const mapStateToProps = (state) => ({
  destination: state.destination,
  pickupLocations: state.pickupLocations,
  currentLocation: state.currentLocation,
  rides: state.rides
});

export const mapDispatchToProps = (dispatch) => ({
  toggleShowing: (pickup) => dispatch(actions.updatePickupShowing(pickup))
});

export default connect(mapStateToProps, mapDispatchToProps)(MapContainer);