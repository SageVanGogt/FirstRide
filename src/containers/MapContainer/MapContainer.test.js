import React from 'react';
import { shallow } from 'enzyme';
import { MapContainer, mapStateToProps, mapDispatchToProps } from './MapContainer';
import * as MOCK from './../../apiCalls/mockData';
import * as API from './../../apiCalls/apiCalls';

jest.mock('./../../apiCalls/apiCalls');

describe('MapContainer', () => {
  let wrapper;
  let mockToggleShowing;
  let mockDestination;

  beforeEach(() => {
    mockToggleShowing = jest.fn();
    mockDestination = {
      id: 1
    };
    wrapper = shallow(<MapContainer 
      destination={mockDestination}
      toggleShowing={mockToggleShowing}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('toggleShowing', () => {
    it('should call toggleShowing with the correct params', () => {
      let expected = {};
      wrapper.instance().toggleShowing({});

      expect(mockToggleShowing).toHaveBeenCalledWith(expected);
    })
  })

  describe('mapStateToProps', () => {

    it('should return the currentLocation prop from state', () => {
      let mockState = {
        destination: {},
        rides: [{}, {}],
        currentLocation: {}
      };
      let mappedProps = mapStateToProps(mockState);
      let expected = mockState.currentLocation;
      let actual = mappedProps.currentLocation;

      expect(actual).toEqual(expected);
    })

    it('should return the pickupLocations from state', () => {
      let mockState = {
        destination: {
          id: 1,
          location_name: 'Red Rocks'
        },
        rides: [{}, {}],
        pickupLocations: MOCK.mockPickups.pickup
      };
      let mappedProps = mapStateToProps(mockState);
      let expected = mockState.pickupLocations;
      let actual = mappedProps.pickupLocations;

      expect(actual).toEqual(expected);
    })

    it('should return the rides from state', () => {
      let mockState = {
        destination: {
          id: 1,
          location_name: 'Red Rocks'
        },
        rides: [{}, {}],
        pickupLocations: MOCK.mockPickups.pickup
      };
      let mappedProps = mapStateToProps(mockState);
      let expected = mockState.rides;
      let actual = mappedProps.rides;

      expect(actual).toEqual(expected);
    })
  })

  describe('mapDispatchToProps', () => {

    it('should call dispatch with the correct params', () => {
      let mockDispatch = jest.fn();
      let mappedProps = mapDispatchToProps(mockDispatch);
      let expected = {
        type: "TOGGLE_SHOWING", 
        pickup: {}
      };
      mappedProps.toggleShowing(expected.pickup);

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });
  })
})