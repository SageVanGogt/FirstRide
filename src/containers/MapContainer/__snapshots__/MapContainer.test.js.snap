// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapContainer
    destination={
        Object {
            "id": 1,
          }
    }
    toggleShowing={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <withScriptjs(withGoogleMap(Component))
        containerElement={
                <div
                        className="availableRidesMap"
                />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBpg9nHLSVUVkDpaTBENL6Q3xC23yWcmco&v=3.exp&libraries=geometry,drawing,places"
        loadingElement={
                <div
                        style={
                                Object {
                                        "height": "100%",
                                      }
                        }
                />
        }
        mapElement={
                <div
                        style={
                                Object {
                                        "height": "100%",
                                      }
                        }
                />
        }
        markerCoords={undefined}
        position={undefined}
        rides={undefined}
        toggleShowing={[Function]}
/>,
      "className": "map-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "containerElement": <div
          className="availableRidesMap"
/>,
        "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyBpg9nHLSVUVkDpaTBENL6Q3xC23yWcmco&v=3.exp&libraries=geometry,drawing,places",
        "loadingElement": <div
          style={
                    Object {
                              "height": "100%",
                            }
          }
/>,
        "mapElement": <div
          style={
                    Object {
                              "height": "100%",
                            }
          }
/>,
        "markerCoords": undefined,
        "position": undefined,
        "rides": undefined,
        "toggleShowing": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <withScriptjs(withGoogleMap(Component))
          containerElement={
                    <div
                              className="availableRidesMap"
                    />
          }
          googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBpg9nHLSVUVkDpaTBENL6Q3xC23yWcmco&v=3.exp&libraries=geometry,drawing,places"
          loadingElement={
                    <div
                              style={
                                        Object {
                                                  "height": "100%",
                                                }
                              }
                    />
          }
          mapElement={
                    <div
                              style={
                                        Object {
                                                  "height": "100%",
                                                }
                              }
                    />
          }
          markerCoords={undefined}
          position={undefined}
          rides={undefined}
          toggleShowing={[Function]}
/>,
        "className": "map-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "containerElement": <div
            className="availableRidesMap"
/>,
          "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyBpg9nHLSVUVkDpaTBENL6Q3xC23yWcmco&v=3.exp&libraries=geometry,drawing,places",
          "loadingElement": <div
            style={
                        Object {
                                    "height": "100%",
                                  }
            }
/>,
          "mapElement": <div
            style={
                        Object {
                                    "height": "100%",
                                  }
            }
/>,
          "markerCoords": undefined,
          "position": undefined,
          "rides": undefined,
          "toggleShowing": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
